src/main/java/com/connexta/poller/service/StatusService.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright (c) 2019 Connexta, LLC
+ *
+ * Released under the GNU Lesser General Public License version 3; see
+ * https://www.gnu.org/licenses/lgpl-3.0.html
+ */
+package com.connexta.poller.service;
+
+import static com.dyngr.core.AttemptResults.continueFor;
+import static com.dyngr.core.AttemptResults.finishWith;
+import static com.dyngr.core.AttemptResults.justContinue;
+
+import com.dyngr.PollerBuilder;
+import com.dyngr.core.AttemptMaker;
+import com.dyngr.core.AttemptResult;
+import com.dyngr.core.StopStrategies;
+import com.dyngr.core.WaitStrategies;
+import java.net.URI;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.HttpStatusCodeException;
+import org.springframework.web.client.ResourceAccessException;
+import org.springframework.web.client.RestTemplate;
+
+/**
+ * This is class is meant to be instantiated as a Bean, Component, or Service. The class's
+ * responsibilities are to manage the settings for polling and kick off new polling tasks. Other
+ * responsibilities include settings the wait and retry behavior. It also creates and configures the
+ * executor, and ensures that it is shutdown properly.
+ */
+@Service
+@Slf4j
+public class StatusService {
+
+  private final RestTemplate restTemplate;
```

Comment:
We should use `WebClient` instead of `RestTemplate` because `RestTemplate` will be deprecated: https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/web/client/RestTemplate.html

Reply:
I will switch it to WebClient.

Reply:
It turns that MockResServiceServer does not work with WebClient the way RestTemplate does. Spring's official advice is to use `OkHttp MockWebServer`.

https://docs.spring.io/spring-framework/docs/current/spring-framework-reference/web-reactive.html#webflux-client-testing

Reply:
Right. Our search tests currently use that.

Reply:
I tried out WebClient, but swtiched back to RestTemplate. It was easier to unit test ResTemplate and also easier to process Spring HTTP exceptions.